src/content/docs/kubernetes-pixie/kubernetes-integration/advanced-configuration/link-otel-applications-kubernetes.mdx

```diff
@@ -0,0 +1,252 @@
+     # Get pod ip so that k8sattributes can tag resources
+- name: POD_NAME 
+valueFrom: 
+fieldRef: 
+fieldPath: metadata.name 
+- name: POD_UID 
+valueFrom: 
+fieldRef: 
+fieldPath: metadata.uid
+      # This is picked up by the resource detector
+    - name: OTEL_RESOURCE_ATTRIBUTES
+      value: service.instance.id=$(POD_NAME),k8s.pod.uid=$(POD_UID)”</var>
+```
+
+## Configure and deploy the OpenTelemetry collector as an agent [#agent]
+
+We recommend you deploy the [collector as an agent](https://opentelemetry.io/docs/collector/getting-started/#agent) on every node within a Kubernetes cluster. The agent can receive telemetry data as well as enrich telemetry data with metadata. For example, the collector can add custom attributes or infrastructure information through processors as well as handle batching, retry, compression and other more advanced features that are handled less efficiently at the client instrumentation level.
+
+For help configuring the collector, see the sample collector configuration file below, along with sections about setting up these options:
+
+  * [OTLP exporter](#otlp-exporter)
+  * [batch processor](#batch)
+  * [resourcedetection processor](#resource-detection)
+  * [k8sattributes processor: general](#attributes-general)
+  * [k8sattributes processor: RBAC](#rbac)
+  * [k8sattributes processor: filters](#discovery-filter)
+
+```yaml
+receivers:
+  otlp:
+    protocols:
+      grpc:
+
+processors:
+  batch:
+  resource:
+    attributes:
+      - key: host.id
+        from_attribute: host.name
+        action: upsert
+  resourcedetection:
+    detectors: [ gke, gce ]
+  k8sattributes:
+    auth_type: "serviceAccount"
+    passthrough: false
+    filter:
+      node_from_env_var: KUBE_NODE_NAME
+    extract:
+      metadata:
+        - k8s.pod.name
+        - k8s.pod.uid
+        - k8s.deployment.name
+        - k8s.cluster.name
+        - k8s.namespace.name
+        - k8s.node.name
+        - k8s.pod.start_time
+    pod_association:
+      - from: resource_attribute
+        name: k8s.pod.uid
+
+exporters:
+  otlp:
+    endpoint: $OTEL_EXPORTER_OTLP_ENDPOINT
+    headers:
+      api-key: $NEW_RELIC_API_KEY
+  logging:
+    logLevel: DEBUG
+
+service:
+  pipelines:
+    metrics:
+      receivers: [ hostmetrics, otlp ]
+      processors: [ resourcedetection, k8sattributes, resource, cumulativetodelta, batch ]
+      exporters: [ otlp ]
+    traces:
+      receivers: [ otlp ]
+      processors: [ resourcedetection, k8sattributes, resource, batch ]
+      exporters: [ otlp ]
+    logs:
+      receivers: [ otlp ]
+      processors: [ resourcedetection, k8sattributes, resource, batch ]
+      exporters: [ otlp ]
+```
+
+### Step 1: Configure the OTLP exporter [#otlp-exporter]
+
+First, configure it by adding an OTLP exporter to your [OpenTelemetry collector configuration YAML file](https://opentelemetry.io/docs/collector/configuration/) along with your New Relic API key as a header.
```

Comment:
Which API key are we talking here? I'm assuming license key, as that's used for ingest, but not sure. The two choices would be either license key (used for ingest) or user key (used for NerdGraph). Whatever we determine, I'd edit the variable to read something like 'YOUR_LICENSE_KEY'. And maybe also clarify in intro paragraph something like 'use the license key for the New RElic account you want to report data to'.

Reply:
@zuluecho9, I clarified this "ingest license key." Also, users need to use this literal environment variable name, so I clarified in the prerequisites that this is the environment variable for the ingest key.